core/src/main/java/io/grpc/Metadata.java

```diff
@@ -490,6 +491,7 @@ private boolean bytesEqual(byte[] left, byte[] right) {
   }
 
   /** Marshaller for metadata values that are serialized into raw binary. */
+  @Immutable
```

Comment:
I can imagine some case where a marshaller might want to be mutable.  For example, the protobuf marshaller may want to cache a copy of the encoding settings.  

What's the reason for wanting to make these things immutable?